src/topics/TopicEditModal.js

```diff
@@ -0,0 +1,172 @@
+// @flow
+import React, { useState, useContext, useMemo } from 'react';
+import { Modal, Text, Pressable, View, TextInput, Platform } from 'react-native';
+import type { JSX$Element } from 'tsflower/subst/react';
+import { ThemeContext, BRAND_COLOR, createStyleSheet } from '../styles';
+import { updateMessage } from '../api';
+import type { Auth, Stream, GetText } from '../types';
+
+type TopicEditModalArgs = {
+  topicModalState: {
+    visible: boolean,
+    topic: string,
+    fetchArgs: {
+      auth: Auth,
+      messageId: number,
+      zulipFeatureLevel: number,
+    },
```

Comment:
I think we don't need to set off this `fetchArgs` property. Is "fetch" a reference to the Fetch API, which we invoke deep down in the `apiFetch` function in src/api/apiFetch.js? If so, there are other things that make their way into that `fetch` call but don't appear here, like this endpoint's `send_notification_to_old_thread` param.

I see we have an existing name `fetchArgs` in the codebase, but I think that's meant to refer to the act of fetching data (messages) from the server. The main thing being done in the edit-topic UI is giving a command _to_ the server, to update its data.